service/internal/proctelemetry/config.go

```diff
@@ -92,3 +138,44 @@ func cardinalityFilter(kvs ...attribute.KeyValue) attribute.Filter {
 		return !filter.HasValue(kv.Key)
 	}
 }
+
+func initPrometheusExporter(prometheusConfig telemetry.Prometheus, asyncErrorChannel chan error) (sdkmetric.Reader, *http.Server, error) {
+	promRegistry := prometheus.NewRegistry()
+	if prometheusConfig.Host == nil {
+		return nil, nil, fmt.Errorf("host must be specified")
+	}
+	if prometheusConfig.Port == nil {
+		return nil, nil, fmt.Errorf("port must be specified")
+	}
+	wrappedRegisterer := prometheus.WrapRegistererWithPrefix("otelcol_", promRegistry)
+	// We can remove `otelprom.WithoutUnits()` when the otel-go start exposing prometheus metrics using the OpenMetrics format
+	// which includes metric units that prometheusreceiver uses to trim unit's suffixes from metric names.
+	// https://github.com/open-telemetry/opentelemetry-go/issues/3468
+	exporter, err := otelprom.New(
+		otelprom.WithRegisterer(wrappedRegisterer),
+		otelprom.WithoutUnits(),
+		// Disabled for the moment until this becomes stable, and we are ready to break backwards compatibility.
+		otelprom.WithoutScopeInfo())
+	if err != nil {
+		return nil, nil, fmt.Errorf("error creating otel prometheus exporter: %w", err)
+	}
+
+	exporter.RegisterProducer(opencensus.NewMetricProducer())
+	return exporter, InitPrometheusServer(promRegistry, fmt.Sprintf("%s:%d", *prometheusConfig.Host, *prometheusConfig.Port), asyncErrorChannel), nil
+}
+
+func initExporter(_ context.Context, exporters telemetry.MetricExporter, asyncErrorChannel chan error) (sdkmetric.Reader, *http.Server, error) {
+	for exporterType, exporter := range exporters {
+		switch exporterType {
+		case prometheueExporter:
+			e, ok := exporter.(telemetry.Prometheus)
+			if !ok {
+				return nil, nil, fmt.Errorf("prometheus exporter invalid: %v", exporter)
```

Comment:
This error message is a bit confusing IMO, should we show the type (`%T`) of the `exporter` here and say the type we expected it to have?